### PR TITLE
[Travel] travel code improvements

### LINKED ISSUE #3872: APIs integration to replace mock data

### LINKED ISSUE #4430: [Travel] Enhance Youtube search results
**Describe the bug**
The current implementation is using "travel" without translation (e.g. "沖繩+travel")

**The expected result**
Use the localized travel term to search Youtube videos (e.g. "沖繩＋旅遊”)

### PR DESCRIPTION
**Pull request by kinse**
Close #3872 
Close #4430

### CODE REVIEW

File: app/src/main/java/org/mozilla/rocket/content/travel/data/TravelRemoteDataSource.kt [hunk 1]
```diff
@@ -190,7 +190,7 @@ class TravelRemoteDataSource : TravelDataSource {
         else
             DEFAULT_VIDEO_ENDPOINT_FORMAT
 
-        return String.format(endPointFormat, String.format(VIDEO_QUERY_PARAM, name), lang)
+        return String.format(endPointFormat, keyword, lang)
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/content/travel/data/TravelRemoteDataSource.kt]: Please help to remove unused `VIDEO_QUERY_PARAM` constant

File: app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelCityViewModel.kt [hunk 2]
```diff
@@ -227,5 +228,6 @@ class TravelCityViewModel(
 
     companion object {
         private const val LOAD_MORE_HOTELS_THRESHOLD = 15
+        private const val VIDEO_QUERY_PARAM = "%s+%s"
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelCityViewModel.kt]: Suggest to call it `VIDEO_QUERY_PATTERN` or `VIDEO_QUERY_PARAM_PATTERN` to be more descriptive.

File: app/src/main/java/org/mozilla/rocket/content/travel/ui/adapter/CityCategoryAdapterDelegate.kt [hunk 3]
```diff
@@ -35,7 +35,14 @@ class CityCategoryViewHolder(
 
     override fun bind(uiModel: DelegateAdapter.UiModel) {
         val cityCategory = uiModel as CityCategoryUiModel
-        category_title.text = cityCategory.subcategoryName
+
+        category_title.text =
+            when (cityCategory.subcategoryName) {
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/content/travel/ui/adapter/CityCategoryAdapterDelegate.kt]: Please follow what we did in shopping and game verticals that use subcategory id to replace the localized category name.
https://github.com/mozilla-tw/FirefoxLite/blob/master/app/src/main/java/org/mozilla/rocket/content/ecommerce/ui/ShoppingMapper.kt#L83

File: app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelMapper.kt [hunk 4]
```diff
@@ -108,10 +107,10 @@ object TravelMapper {
                 ig.linkUrl
             )
 
-    fun toExploreWikiUiModel(wiki: Wiki): WikiUiModel =
+    fun toExploreWikiUiModel(wiki: Wiki, source: String): WikiUiModel =
             WikiUiModel(
                 wiki.imageUrl,
-                SOURCE_WIKI,
```
Comment (benjamin-cheng) [inline app/src/main/java/org/mozilla/rocket/content/travel/ui/TravelMapper.kt]: We do need a property to keep non-translated source name. Please add a new property in `WikiUiModel` to show translated content

**Comment by kinse** [comment]
@benjamin-cheng I saw Harly has prepared new strings but not merged yet. Shall we wait for his merge to finalize the string ?

**Pull request merged**